File: implementation/sipamato/src/test/java/ch/difty/sipamato/persistance/jooq/search/SearchOrderRecordMapperTest.java
package ch.difty.sipamato.persistance.jooq.search;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.when;

import org.jooq.RecordMapper;

import ch.difty.sipamato.db.tables.records.SearchOrderRecord;
import ch.difty.sipamato.entity.SearchOrder;
import ch.difty.sipamato.persistance.jooq.RecordMapperTest;

public class SearchOrderRecordMapperTest extends RecordMapperTest<SearchOrderRecord, SearchOrder> {

    static final Long ID = 2l;
    static final Integer OWNER = 1;
    static final boolean GLOBAL = true;

    public static void entityFixtureWithoutIdFields(SearchOrder entityMock) {
        when(entityMock.getId()).thenReturn(ID);
        when(entityMock.getOwner()).thenReturn(OWNER);
        when(entityMock.isGlobal()).thenReturn(GLOBAL);
    }

    @Override
    protected RecordMapper<SearchOrderRecord, SearchOrder> getMapper() {
        return new SearchOrderRecordMapper();
    }

    @Override
    protected SearchOrderRecord makeRecord() {
        SearchOrderRecord record = new SearchOrderRecord();
        record.setId(ID);
        record.setOwner(OWNER);
        record.setGlobal(GLOBAL);
        return record;
    }

    @Override
    protected void assertEntity(SearchOrder entity) {
        assertThat(entity.getId()).isEqualTo(ID);
        assertThat(entity.getOwner()).isEqualTo(OWNER);
        assertThat(entity.isGlobal()).isEqualTo(GLOBAL);
    }

}

File: core/core-persistence-jooq/src/test/kotlin/ch/difty/scipamato/core/persistence/search/SearchOrderRecordMapperTest.kt
package ch.difty.scipamato.core.persistence.search

import ch.difty.scipamato.core.db.tables.records.SearchOrderRecord
import ch.difty.scipamato.core.entity.search.SearchOrder
import ch.difty.scipamato.core.persistence.RecordMapperTest
import io.mockk.every
import org.amshove.kluent.shouldBeEmpty
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeFalse
import org.jooq.RecordMapper

internal class SearchOrderRecordMapperTest : RecordMapperTest<SearchOrderRecord, SearchOrder>() {

    override val mapper: RecordMapper<SearchOrderRecord, SearchOrder> = SearchOrderRecordMapper()

    override fun makeRecord(): SearchOrderRecord {
        return SearchOrderRecord().apply {
            id = ID
            name = NAME
            owner = OWNER
            global = GLOBAL
        }
    }

    override fun setAuditFieldsIn(record: SearchOrderRecord) {
        record.created = CREATED
        record.createdBy = CREATED_BY
        record.lastModified = LAST_MOD
        record.lastModifiedBy = LAST_MOD_BY
        record.version = VERSION
    }

    override fun assertEntity(entity: SearchOrder) {
        entity.id shouldBeEqualTo ID
        entity.name shouldBeEqualTo NAME
        entity.owner shouldBeEqualTo OWNER
        entity.isGlobal shouldBeEqualTo GLOBAL

        entity.searchConditions.shouldBeEmpty()

        // not persisted and therefore always false
        entity.isShowExcluded.shouldBeFalse()
    }

    companion object {
        const val ID = 2L
        const val NAME = "soName"
        const val OWNER = 1
        const val GLOBAL = true

        fun entityFixtureWithoutIdFields(entityMock: SearchOrder) {
            every { entityMock.name } returns NAME
            every { entityMock.owner } returns OWNER
            every { entityMock.isGlobal } returns GLOBAL

            auditFixtureFor(entityMock)
        }
    }
}
